src/Spago/Build.hs

```diff
@@ -323,11 +323,14 @@ runBackend
 runBackend maybeBackend RunDirectories{ sourceDir, executeDir } moduleName maybeSuccessMessage failureMessage extraArgs = do
   logDebug $ display $ "Running with backend: " <> fromMaybe "nodejs" maybeBackend
   BuildOptions{ pursArgs } <- view (the @BuildOptions)
-  isES <- Purs.hasMinPursVersion "0.15.0"
+  isES <- Purs.hasMinPursVersion "0.15.0-alpha-01"
```

Comment:
Can `Purs.hasMinPursVersion` be updated to drop the pre-release and other build metadata on the purs version before checking if it matches the minimum version? In other words, could this could be updated to read:
```
isEs <- Purs.hasMinPursVersion "0.15.0"
```

Reply:
I “downgraded” it to work with the current release. "0.15.0" > "0.15.0-alpha-01”.

Reply:
Oh sorry I misread your comment. I guess so, but is this always wanted? I mean `hasMinPursVersion` is also used in other places.

Reply:
When we update the `contrib` libraries, they need to use `spago`. There may be additional alpha releases between now and then, so it might be `alpha-04` by that point. I'd prefer that we don't need to make a new `spago` release bumping the alpha number to something else each time a new PS alpha release is made.

Reply:
I think this shouldn't be a problem, because "0.15.0-alpha-04” should be greater than "0.15.0-alpha-01”, but I will check that this works

Reply:
```
(>=) <$> Version.semver "0.15.0-alpha-04" <*> Version.semver "0.15.0-alpha-01"
Right True
```

Reply:
Is it also the case that `0.15.0 > 0.15.0-alpha-01`?

Reply:
Yes

Reply:
I still think the overall approach here is still problematic. This line should read:
```haskell
isES <- Purs.hasMinPursVersion "0.15.0"
```
or better yet
```hsakell
isES <- Purs.hasMinPursVersion purs0_15_0
where
-- purs0_15_0 would be defined as a constant somewhere
purs0_15_0 = SemVer 0 15 0 [] Nothing
```

The problem with the above line is that we don't indicate whether the possible pre-release and metadata stuff should be dropped in the `hasMinPurs` comparsion check. So, perhaps that function could take a second argument that indicates whether such "extra" information should be ignored in the comparison? If so, then it might read:
```
data SemVerComparison
   = IncludePreAndMeta
   | IgnorePreAndMeta

isES <- Purs.hasMinPursVersion purs0_15_0 IgnorePreAndMeta
```
and updating `hasMinPursVersion` as needed.

Reply:
Regarding your first proposal:
I can do this, but then the `-alpha-0x` versions won't work.
Regarding your second proposal: I am not really sure how to do this.

Reply:
This is what I mean: https://github.com/working-group-purescript-es/spago/pull/2

Reply:
@JordanMartinez I am not sure I understand why including the prerelease segment is problematic there? This feature is indeed supported starting from the `alpha` releases, so this looks correct to me?

Reply:
@f-f @sigma-andex With how things are currently implemented, would all of the below expressions evaluate to true?
```
foo = do
  let
    s = Version.semver
    gt left right = (>) <$> left <*> right
  s "0.15.0-alpha-01" `gt `s "0.14.7"
  s "0.15.0" `gt` s "0.15.0-alpha-01"
  s "0.15.0" `gt` s "0.15.0-alpha-02"
  s "0.15.0" `gt` s "0.15.0-rc-01"
  s "0.15.0" `gt` s "0.15.0-rc-02"
```
If so, then I will drop this point.

Reply:
```
>  s "0.15.0-alpha-01" `gt `s "0.14.7"
Right True
>  s "0.15.0" `gt` s "0.15.0-alpha-01"
Right True
>  s "0.15.0" `gt` s "0.15.0-alpha-02"
Right True
>  s "0.15.0" `gt` s "0.15.0-rc-01"
Right True
>  s "0.15.0" `gt` s "0.15.0-rc-02"
Right True
```

Reply:
Welp! Ignore me! :sweat_smile: